internal/internal_worker_base.go

```diff
@@ -288,8 +291,9 @@ func (bw *baseWorker) pollTask() {
 		}
 	}
 
-	bw.retrier.Throttle()
+	bw.retrier.Throttle() // sleeps if retry policy determines it should sleep after failures
 	if bw.pollLimiter == nil || bw.pollLimiter.Wait(bw.limiterContext) == nil {
+		// TODO: block here on a shared semaphore with history-loading?
```

Comment:
this is where I'm planning on putting it.

at this point the error-throttler has already delayed things, and we're inside both the auto-scaler and per-second limiter, so an _immediate_ poll is correct, and a _delayed_ poll is... potentially exceeding per-second, but not exceeding concurrent.

this is not the most hygienic location, I think this whole thing might be worth revamping.  but I don't believe it will make it _behave_ worse in the meantime.